src/sc/sidechaintypes.cpp

```diff
@@ -1,139 +1,329 @@
+    return res;
 }
+#endif
+////////////////////////////// End of Field types //////////////////////////////
 
-void CompressedFieldElement::InitFieldElement() const
+/////////////////////// libzendoomc namespace definitions //////////////////////
+namespace libzendoomc
 {
-    if (scFieldElement.IsNull())
+
+    bool IsValidScProof(const ScProof& scProof)
     {
-        if (vRawField.size() > scFieldElement.size())
-        {
-            LogPrint("sc", "%s():%d - Error: Wrong size: rawData[%d]>fieldElement[%d]\n", 
-                __func__, __LINE__, vRawField.size(), scFieldElement.size());
-            throw std::invalid_argument(
-                std::string("Wrong size: rawData[" + std::to_string(vRawField.size()) + "] > fieldElement[" +
-                std::to_string(scFieldElement.size()) + "]"));
-        }
-        // pad with zeroes, must have the same size for the internal repr
-        std::vector<unsigned char> temp(vRawField);
-        temp.resize(scFieldElement.size(), 0x0);
-        *const_cast<libzendoomc::ScFieldElement*>(&scFieldElement) = libzendoomc::ScFieldElement(temp); // TODO
+        auto scProofDeserialized = zendoo_deserialize_sc_proof(scProof.begin());
+        if (scProofDeserialized == nullptr)
+            return false;
+        zendoo_sc_proof_free(scProofDeserialized);
+        return true;
+    }
+
+    bool IsValidScVk(const ScVk& scVk)
+    {
+        auto scVkDeserialized = zendoo_deserialize_sc_vk(scVk.begin());
+        if (scVkDeserialized == nullptr)
+            return false;
+        zendoo_sc_vk_free(scVkDeserialized);
+        return true;
+    }
+
+    std::string ToString(Error err){
+        return strprintf(
+            "%s: [%d - %s]\n",
+            err.msg,
+            err.category,
+            zendoo_get_category_name(err.category));
     }
 }
+/////////////////// end of libzendoomc namespace definitions ///////////////////
 
-const libzendoomc::ScFieldElement& CompressedFieldElement::GetFieldElement() const
+////////////////////////////// Custom Config types //////////////////////////////
+bool FieldElementCertificateFieldConfig::IsValid() const
 {
-    InitFieldElement();
-    return scFieldElement;
+    if(nBits <=0 || nBits > SC_FIELD_SIZE*8)
+        return false;
+    else
+        return true;
 }
 
-#ifdef BITCOIN_TX
-bool CompressedFieldElement::IsValid() const { return true; }
-#else
-bool CompressedFieldElement::IsValid() const
+FieldElementCertificateFieldConfig::FieldElementCertificateFieldConfig(int32_t nBitsIn): CustomCertificateFieldConfig(), nBits(nBitsIn) {}
+
+int32_t FieldElementCertificateFieldConfig::getBitSize() const
+{
+    return nBits;
+}
+
+//----------------------------------------------------------------------------------
+bool BitVectorCertificateFieldConfig::IsValid() const
 {
-    InitFieldElement();
-    if (scFieldElement.IsNull())
+    bool isBitVectorSizeValid = (bitVectorSizeBits >= 0) && (bitVectorSizeBits <= MAX_BIT_VECTOR_SIZE_BITS);
```

Comment:
is it fine to have `bitVectorSizeBits == 0`?

Reply:
No. Changed